controllers/managedfusion_controller.go

```diff
@@ -689,6 +723,135 @@ func (r *ManagedFusionReconciler) reconcilePrometheusProxyNetworkPolicy() error
 	return nil
 }
 
+func (r *ManagedFusionReconciler) reconcileEgressFirewall() error {
+	if !r.AvailableCRDs[EgressFirewallCRD] {
+		r.Log.Info("EgressFirewall CRD not present, skipping reconcile for egress firewall")
+		return nil
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressFirewall, func() error {
+		if err := r.own(r.egressFirewall); err != nil {
+			return err
+		}
+		desiredEgressRules, err := r.getEgressRules()
+		if err != nil {
+			return err
+		}
+		egressFirewallRules := []ovnv1.EgressFirewallRule{}
+		for i := range desiredEgressRules {
+			egressFirewallRules = append(
+				egressFirewallRules,
+				ovnv1.EgressFirewallRule{
+					Type: ovnv1.EgressFirewallRuleType(desiredEgressRules[i].ruleType),
+					To: ovnv1.EgressFirewallDestination{
+						CIDRSelector: desiredEgressRules[i].cidrSelector,
+						DNSName:      desiredEgressRules[i].dnsName,
+					},
+				},
+			)
+		}
+		desired := templates.EgressFirewallTemplate.DeepCopy()
+		desired.Spec.Egress = append(egressFirewallRules, desired.Spec.Egress...)
+		r.egressFirewall.Spec = desired.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update egressFirewall: %v", err)
+	}
+	return nil
+
+}
+
+func (r *ManagedFusionReconciler) reconcileEgressNetworkPolicy() error {
+	if !r.AvailableCRDs[EgressNetworkPolicyCRD] {
+		r.Log.Info("EgressNetworkPolicy CRD not present, skipping reconcile for egress network policy")
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressNetworkPolicy, func() error {
+		if err := r.own(r.egressNetworkPolicy); err != nil {
+			return err
+		}
+		desiredEgressRules, err := r.getEgressRules()
+		if err != nil {
+			return err
+		}
+		egressFirewallRules := []openshiftv1.EgressNetworkPolicyRule{}
+		for i := range desiredEgressRules {
+			egressFirewallRules = append(
+				egressFirewallRules,
+				openshiftv1.EgressNetworkPolicyRule{
+					Type: openshiftv1.EgressNetworkPolicyRuleType(desiredEgressRules[i].ruleType),
+					To: openshiftv1.EgressNetworkPolicyPeer{
+						CIDRSelector: desiredEgressRules[i].cidrSelector,
+						DNSName:      desiredEgressRules[i].dnsName,
+					},
+				},
+			)
+		}
+		desired := templates.EgressNetworkPolicyTemplate.DeepCopy()
+		desired.Spec.Egress = append(egressFirewallRules, desired.Spec.Egress...)
+		r.egressNetworkPolicy.Spec = desired.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update egressNetworkPolicy: %v", err)
+	}
+	return nil
+}
+
+type egressRules struct {
+	ruleType     string
+	dnsName      string
+	cidrSelector string
```

Comment:
- At usage site, pls make sure only one of these are set as these are mutually exclustive
- If an empty string is considered unset then we are fine

Reply:
This is not relevant anymore as we have another approach to solve the problem.